storage/database/src/backend/redb/env.rs

```diff
@@ -189,7 +189,10 @@ where
         // 3. So it's not being used to open a table since that needs `&tx_rw`
         //
         // Reader-open tables do not affect this, if they're open the below is still OK.
-        redb::WriteTransaction::delete_table(tx_rw, table)?;
+        if !redb::WriteTransaction::delete_table(tx_rw, table)? {
+            return Err(RuntimeError::TableNotFound);
+        }
```

Comment:
Unrelated: fixed `redb`'s `clear_db` behavior. It is supposed to error if the table didn't exist, test added...